atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -83,15 +85,35 @@ private synchronized void registerMetricWithFqn(String fullyQualifiedMetricName,
         }
     }
 
-    public Meter registerMeter(Class clazz, String meterName) {
-        return registerMeter(MetricRegistry.name(clazz, "", meterName));
+    public Histogram registerOrGetHistogram(Class clazz, String metricName) {
+        return registerOrGetHistogram(MetricRegistry.name(clazz, metricName));
     }
 
-    public Meter registerMeter(Class clazz, String metricPrefix, String meterName) {
-        return registerMeter(MetricRegistry.name(clazz, metricPrefix, meterName));
+    private Histogram registerOrGetHistogram(String fullyQualifiedHistogramName) {
+        Histogram histogram = metricRegistry.histogram(fullyQualifiedHistogramName);
+        registeredMetrics.add(fullyQualifiedHistogramName);
```

Comment:
Why are we keeping our own set of registered metrics? it looks like right now it’s a regular HashSet that would require us to synchronize all reads & writes on these registerOrGet methods. We should switch to a concurrent hash set